app/src/main/java/org/oppia/android/app/story/storyitemviewmodel/StoryChapterSummaryViewModel.kt

```diff
@@ -22,6 +22,7 @@ class StoryChapterSummaryViewModel(
   val topicId: String,
   val storyId: String,
   val chapterSummary: ChapterSummary,
+  val previousChapterTitle: String,
```

Comment:
While it's true that currently the previous chapter is always blocking the next one, I think it would be a bit cleaner if we pass along the computed blocking chapter summary (as a ``ChapterSummary`` object) & reference its name in XML in the same way we do for the summary of this card.

I think to do that we will need to add a 'prerequisite_chapter' field to the summary proto: https://github.com/oppia/oppia-android/blob/74db288b9ac25fb942a8feb35e918bcd7e94e55c/model/src/main/proto/topic.proto#L86 & pipe it from the domain controller (https://github.com/oppia/oppia-android/blob/2c02e06d768d5341569e326d9ec042c91c3586da/domain/src/main/java/org/oppia/android/domain/topic/TopicListController.kt).

This also solves another part I'm concerned with: it provides a well-defined default if there is no blocking chapter (default ChapterSummary proto).

Note that ideally we would model this as a oneof in ChapterSummary, but I think that could be filed as a future issue for someone else to take on since it will require a broader refactoring.

Reply:
I don't think I know how to "pipe it from the domain controller"

Reply:
PTAL, if that's what you had in mind before I add tests.

Reply:
Tried to debug the app and It seems like `prerequisiteChapter` is always containing the null items. Also, I tried running your PR, The name of the previous chapter is not visible. 

I will look into this in detail later today my time. 

![Screenshot 2020-12-11 at 09 39 52](https://user-images.githubusercontent.com/25057618/101861649-8bb59b00-3b96-11eb-8bff-2d9af173d160.png)

Reply:
`missingPrerequisiteChapter` is only present in locked chapters. So I guess, in some cases it's intended.

Reply:
I added setting `missingPrerequisiteChapter` into one more function,I think that was the problem. PTAL

Reply:
Ah, nice catch. @viktoriias can you double check that the new tests you added in TopicListController cover both functions? A quick check could be to comment out each of the changes in TopicListController one-at-a-time to verify that one or more of the tests changes fail.

Reply:
Tests were failing, that's why I had to look for another place to initialize the `missingPrerequisiteChapter`.

> > > Thanks @viktoriias. Overall, this looks great! I have one nit, and one main comment: you added a test for TopicControllerTest, but can you also add a UI-facing test to make sure the binding logic in story_chapter_view is tested? I think that would require a new test in [StoryFragmentTest](https://github.com/oppia/oppia-android/blob/develop/app/src/sharedTest/java/org/oppia/android/app/story/StoryFragmentTest.kt). Please add a test for both portrait & landscape (the latter can be done by changing orientation--see `testStoryFragment_changeConfiguration_chapterSummaryIsShownCorrectly`).
> > 
> > 
> > I looked into this test file initially, per @anandwana001's advice. It marks first story as completed, so prerequisite didn't show up. I'll need to remove `markPartialStoryProgressForFractions` to be able to test summary for the second story.
> 
> Removing `markPartialStoryProgressForFractions ` might cause failure for other test. I think what we can do here is to shift it from the `@Before` to all the required test before launching the test. This way we can have test specific weather or not we require partial progress.

I only affects `testStoryFragment_correctStoryCountLoadedInHeader` test: `1 of 2 Chapters Completed` -> `0 of 2 Chapters Completed`. I think it's fine to remove it, let me know what you think.

Reply:
The tests I added were failing at first, so I had to look for another place to initialize `missingPrerequisiteChapter`.

Reply:
Tests in TopicListController don't fully cover `combineTopicAndTopicProgress` and `combineStorySummaryAndStoryProgress`. In particular, they don't cover my original change, I only got visual confirmation when running the app. I'm not sure how to get to that particular scenario in `combineStorySummaryAndStoryProgress`..

Reply:
Good catch. I think we should go ahead and add a new test to cover the other branch. I think the branch that's being missed is indicating that our tests aren't actually completing a story before retrieving the chapter list. Could we adapt the test that fails when the other branch is commented out to cover this case?

Reply:
Done. I added one more test (and verified it can fail).

Reply:
Awesome--the new test looks good!